src/main/scala/akka/stream/alpakka/xlsx/CellReference.scala

```diff
@@ -7,68 +7,68 @@ import akka.stream.alpakka.xml.Attribute
 import scala.annotation.tailrec
 import scala.util.Try
 
-case class CellReference(name: String, colNum: Int, rowNum: Int)
+final case class CellReference(fullName: String, columnIndex: Int, rowIndex: Int)
 
 object CellReference {
+  private final val BaseLettersNum = ('A' to 'Z').size
+  private final val AsciiCodeA     = 'A'
+  private final val AsciiCodeZ     = 'Z'
 
-  private val CELL_REF = """(\s+"""
 
-  private def convertColStringToIndex(ref: String): Int = {
-    @tailrec
-    def solver(refArray: List[Char], index: Int, retval: Int): Int = {
-      refArray.headOption match {
-        case Some(c) =>
-          if (index != 0) throw new IllegalArgumentException("Bad col ref format '" + ref + "'")
-          else solver(refArray.tail, index + 1, retval * 26 + c - 65 + 1)
-        case None => retval
-      }
+  private[xlsx] def parseReference(attributes: List[Attribute]): Option[CellReference] = {
+    def splitCellReference(reference: String) = {
+      val referenceLength       = reference.length
+      val referenceColumn       = reference.filterNot(c => c >= '0' && c <= '9')
+      val referenceColumnLength = referenceColumn.length
+
+      if (referenceLength == 0 || referenceColumnLength == 0 || referenceColumnLength == referenceLength) None
+      else Some((reference, referenceColumn, reference.substring(referenceColumnLength)))
     }
 
-    solver(ref.toUpperCase(Locale.ROOT).toCharArray.toList, 0, 0)
-  }
+    def convertColumnStringToIndex(column: String) = {
+      @tailrec
+      def convert(referencePart: String, currentIndex: Int): Int = {
+        referencePart.headOption match {
+          case Some(char) if AsciiCodeA <= char && char <= AsciiCodeZ =>
+            // `char - AsciiCodeA` converts base capital letters to consecutive numbers (starting from 0), ie. 'A' -> 0, 'B' -> 1, ...
+            convert(referencePart.tail, currentIndex * BaseLettersNum + (char - AsciiCodeA) + 1)
+          case Some(invalidLetterNum) =>
+            throw new IllegalArgumentException(s"Unexpected letter of code $invalidLetterNum found.")
+          case None => currentIndex
+        }
+      }
 
-  private def splitCellRef(ref: String) = {
-    val len  = ref.length
-    val s1   = ref.filterNot(c => c >= '0' && c <= '9')
-    val sLen = s1.length
-    if (len == 0 || sLen == len) {
-      None
-    } else {
-      Some((ref, s1, ref.substring(len - sLen)))
+      convert(column.toUpperCase(Locale.ROOT), 0)
     }
-  }
 
-  private[xlsx] def parseRef(attrs: List[Attribute]): Option[CellReference] = {
-    attrs
-      .find(_.name == "r")
-      .flatMap { attr =>
-        splitCellRef(attr.value)
-      }
-      .flatMap {
-        case (ref, s1, s2) =>
-          Try(CellReference(ref, convertColStringToIndex(s1), Integer.parseInt(s2))).toOption
+    def convertRowStringToIndex(row: String) = Try(row.toInt).filter(_ > 0).get
```

Comment:
You'd probably be better off (refactoring-wise) returning a `Try` here and only mapping over it below.